LayoutTests/imported/w3c/web-platform-tests/mixed-content/gen/top.meta/unset/img-tag.https-expected.txt

```diff
@@ -1,13 +1,10 @@
-Blocked access to external URL http://www1.localhost:8800/common/security-features/subresource/image.py?redirection=keep-scheme&action=purge&key=GENERATED_KEY&path=%2Fmixed-content
-Blocked access to external URL http://www1.localhost:8800/common/security-features/subresource/image.py?redirection=no-redirect&action=purge&key=GENERATED_KEY&path=%2Fmixed-content
-Blocked access to external URL http://www1.localhost:8800/common/security-features/subresource/image.py?redirection=swap-scheme&action=purge&key=GENERATED_KEY&path=%2Fmixed-content
 
 FAIL Mixed-Content: Expects allowed for img-tag to cross-http origin and keep-scheme redirection from https context. assert_equals: The resource request should be 'allowed'. expected "allowed" but got "blocked"
 FAIL Mixed-Content: Expects allowed for img-tag to cross-http origin and no-redirect redirection from https context. assert_equals: The resource request should be 'allowed'. expected "allowed" but got "blocked"
 FAIL Mixed-Content: Expects allowed for img-tag to cross-http origin and swap-scheme redirection from https context. assert_equals: The resource request should be 'allowed'. expected "allowed" but got "blocked"
-PASS Mixed-Content: Expects allowed for img-tag to same-http origin and keep-scheme redirection from https context.
-PASS Mixed-Content: Expects allowed for img-tag to same-http origin and no-redirect redirection from https context.
-PASS Mixed-Content: Expects allowed for img-tag to same-http origin and swap-scheme redirection from https context.
+FAIL Mixed-Content: Expects allowed for img-tag to same-http origin and keep-scheme redirection from https context. assert_equals: The resource request should be 'allowed'. expected "allowed" but got "blocked"
+FAIL Mixed-Content: Expects allowed for img-tag to same-http origin and no-redirect redirection from https context. assert_equals: The resource request should be 'allowed'. expected "allowed" but got "blocked"
+FAIL Mixed-Content: Expects allowed for img-tag to same-http origin and swap-scheme redirection from https context. assert_equals: The resource request should be 'allowed'. expected "allowed" but got "blocked"
```

Comment:
Any idea what's going on here?

Reply:
Yes, this is failure in the testing framework, and I haven't been able to find a good way to fix it (but I will, with some more time). In short, we need to set the appropriate default ports for http/https so that we can correctly upgrade the requests, but this needs to happen before the main document is loaded. If we don't set this before the main document is loaded, then the document's SecurityOrigin will include the port number (e.g., 9443, the port number is set to `std::nullopt` in the SecurityOrigin constructor if it's the default value for the protocol), subsequently when we process the subresource request, we check if a CORS preflight is needed. At this point, a SecurityOrigin is created for the request (and that SecurityOrigin will not include the port number because now we changed the internal default value), and `SecurityOrigin::isSameSchemeHostPort` will fail between the document's SecurityOrigin and the subresource request due to different port values. Therefore, setting the default port in javascript is too late.

These test failures ("blocked" instead of "allowed") is the result of not changing the default port values, so the requests are "upgraded" to `https://localhost:8800`. If we change the default values in javascript, then we get a CORS access control error like `FAIL Mixed-Content: Expects allowed for img-tag to cross-http origin and keep-scheme redirection from https context. promise_test: Unhandled rejection with value: object "Error: Network Error"`

Reply:
Actually, there's even more fallout from this, like `LayoutTests/imported/w3c/web-platform-tests/html/dom/idlharness.https_exclude=(Document_Window_HTML._)-expected.txt`. I'll find a fix for this.

Reply:
Ah, no, I was wrong about that. The default-port issue was masking other problems, but it seems like this one is actually expected. Chrome and Firefox have the same behavior: https://wpt.fyi/results/mixed-content/gen/top.meta/unset/img-tag.https.html?label=master&label=experimental&aligned

WebKit was behaving differently before this change because the the mixed content checker basically allowed all (insecure) images. I updated it so it aligns with the spec, and in this patch we block any images that are `crossorigin`/`CORS` or are part of an imageset. In this specific test, the images are requested via http with CORS, so they are blocked.

Reply:
If you block `<picture>` as well then you can close https://bugs.webkit.org/show_bug.cgi?id=179049